src/Controller.sol

```diff
@@ -0,0 +1,220 @@
+contract Controlled
+{
+	bool private initialized = false;
+	address public owner;
+	Controller internal controller;
+
+	modifier whitelistOnly
+	{
+		controller.assertIsWhitelisted(msg.sender);
+		_;
+	}
+
+	modifier controllerOnly
+	{
+		require(Controller(msg.sender) == controller);
+		_;
+	}
+
+	modifier ownerOnly
+	{
+		require(msg.sender == owner);
+		_;
+	}
+
+	modifier onlyInGoodTimes
+	{
+		controller.stopInEmergency();
+		_;
+	}
+
+	modifier onlyInBadTimes
+	{
+		controller.onlyInEmergency();
+		_;
+	}
+
+	function Controlled()
+	{
+		owner = msg.sender;
+	}
+
+	// TODO: convert into constructor call
+	function initialize(Controller _controller) external ownerOnly returns(bool)
+	{
+		require(initialized == false);
+		initialized = true;
+		controller = _controller;
+		owner = address(0);
+		return true;
+	}
+
+	function setController(Controller _controller) public controllerOnly returns(bool)
+	{
+		controller = _controller;
+		return true;
+	}
+
+	function suicideFunds(address _target) public controllerOnly returns(bool)
+	{
+		selfdestruct(_target);
+		return true;
+	}
+}
+
+contract Controller
+{
+	address public owner;
+	mapping(address => bool) public whitelist;
+	mapping(bytes32 => address) public registry;
+	bool public stopped = false;
+
+	modifier whitelistOnly
+	{
+		assertIsWhitelisted(msg.sender);
+		_;
+	}
+
+	modifier devModeOwnerOnly
+	{
+		require(msg.sender == owner);
+		require(whitelist[owner]);
+		_;
+	}
+
+	modifier ownerOnly
+	{
+		require(msg.sender == owner);
+		_;
+	}
+
+	modifier onlyInBadTimes
+	{
```

Comment:
I think `onlyInGoodTimes` and `onlyInBadTimes` is fine actually. Easy to understand.